### PR TITLE
RavenDB-22295 - UnobservedTaskException  System.ObjectDisposedException

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22295/UnobservedTaskException-System.ObjectDisposedException-Cannot-access-a-disposed-object.-Object-name-Leader-A-in-term-745754.

### Additional description

Unobserved ObjectDisposedException after leader dispose.
the exception of the faulted task that exchanges with _topologyModification on Leader.Dispose

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [ ] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [x] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Rachis/Leader.cs [hunk 1]
```diff
@@ -817,6 +819,19 @@ public void Dispose()
             }
         }
 
+        void ObserveTask(Task task)
```
Comment (karmeli87) [inline src/Raven.Server/Rachis/Leader.cs]: better to use the `IgnoreUnobservedExceptions` extension

Comment (shaharhikri) [reply src/Raven.Server/Rachis/Leader.cs]: Done

File: test/SlowTests/Issues/RavenDB-22295.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public async Task UnobservedObjectDisposedExceptionAfterLeaderDispose()
+    {
+        Exception e = null;
+        EventHandler<UnobservedTaskExceptionEventArgs> handler = (sender, args) =>
+        {
+            if(e is ObjectDisposedException)
+                e = args.Exception;
+        };
+
+        TaskScheduler.UnobservedTaskException += handler;
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-22295.cs]: will this catch exceptions from other tests? is that okay?

Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-22295.cs]: Done

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**